src/components/cesium_map/api/spatial.js

```diff
@@ -604,33 +606,52 @@ export class ISamplesSpatial {
     return this.viewer.dataSources.remove(dataSource, destroy);
   }
 
+  d(rd) {
+    return Cesium.Math.toDegrees(rd);
+  }
+
+  r2str(r) {
+    let x0 = this.d(r.west);
+    let x1 = this.d(r.east);
+    let y0 = this.d(r.south);
+    let y1 = this.d(r.north);
+    return `${x0},${y0},${x1},${y1}`;
+}
+
   gridTracker(viewer, gridder){
+    if (!this.displayGrid) return; 
     let scratchRectangle = new Cesium.Rectangle();
     let rect = viewer.camera.computeViewRectangle(viewer.scene.globe.ellipsoid, scratchRectangle);
+    if (this.r2str(rect) === this.gridder.global_grid.rect_str){
+      return; // no need to update 
```

Comment:
Optmizing the update by only calling update and requests to the server when the rect changes.